nft-minter/src/nft_module.rs

```diff
@@ -0,0 +1,263 @@
+
+use crate::{
+    common_storage::{BrandId, BrandInfo, Tag},
+    unique_id_mapper::{UniqueId, UniqueIdMapper},
+};
+
+const NFT_AMOUNT: u32 = 1;
+const NFT_ISSUE_COST: u64 = 50_000_000_000_000_000; // 0.05 EGLD
+const ROYALTIES_MAX: u32 = 10_000; // 100%
+const VEC_MAPPER_FIRST_ITEM_INDEX: usize = 1;
+
+const MAX_BRAND_ID_LEN: usize = 50;
+static INVALID_BRAND_ID_ERR_MSG: &[u8] = b"Invalid Brand ID";
+
+#[elrond_wasm::module]
+pub trait NftModule:
+    crate::common_storage::CommonStorageModule
+    + crate::admin_whitelist::AdminWhitelistModule
+    + crate::nft_attributes_builder::NftAttributesBuilderModule
+    + crate::royalties::RoyaltiesModule
+{
+    #[payable("EGLD")]
+    #[endpoint(issueTokenForBrand)]
+    fn issue_token_for_brand(
+        &self,
+        brand_id: BrandId<Self::Api>,
+        media_type: ManagedBuffer,
+        royalties: BigUint,
+        max_nfts: usize,
+        mint_start_timestamp: u64,
+        mint_price_token_id: TokenIdentifier,
+        mint_price_amount: BigUint,
+        token_display_name: ManagedBuffer,
+        token_ticker: ManagedBuffer,
+        #[var_args] tags: MultiValueEncoded<Tag<Self::Api>>,
+    ) {
+        self.require_caller_is_admin();
+
+        let id_len = brand_id.len();
+        require!(
+            id_len > 0 && id_len <= MAX_BRAND_ID_LEN,
+            INVALID_BRAND_ID_ERR_MSG
+        );
+
+        let payment_amount = self.call_value().egld_value();
+        require!(
+            payment_amount == NFT_ISSUE_COST,
+            "Invalid payment amount. Issue costs exactly 0.05 EGLD"
+        );
+
+        require!(
+            self.is_supported_media_type(&media_type),
+            "Invalid media type"
+        );
+        require!(royalties <= ROYALTIES_MAX, "Royalties cannot be over 100%");
+        require!(max_nfts > 0, "Cannot create brand with max 0 items");
+        require!(
+            mint_price_token_id.is_egld() || mint_price_token_id.is_valid_esdt_identifier(),
+            "Invalid price token"
+        );
+
+        let is_new_brand = self.registered_brands().insert(brand_id.clone());
+        require!(is_new_brand, "Brand already exists");
+
+        let id_offset = self.last_item_id().update(|last_id| {
+            let prev_last_id = *last_id;
+            *last_id += max_nfts;
+            require!(*last_id > prev_last_id, "ID overflow!");
+
+            prev_last_id
+        });
+        let brand_info = BrandInfo {
+            token_display_name: token_display_name.clone(),
+            media_type,
+            id_offset,
+            royalties,
+            mint_start_timestamp,
+            mint_price_token_id,
+            mint_price_amount,
+        };
+        self.brand_info(&brand_id).set(&brand_info);
+        self.available_ids(&brand_id).set_initial_len(max_nfts);
+
+        if !tags.is_empty() {
+            self.tags_for_brand(&brand_id).set(&tags.to_vec());
+        }
+
+        self.nft_token(&brand_id).issue(
```

Comment:
use the new endpoint of issue and set all roles :D

Reply:
but it can stay as it is.

Reply:
The new function will be available in this mapper in the next framework version, will change then.